persistence/remote/src/main/java/org/infinispan/persistence/remote/upgrade/HotRodTargetMigrator.java

```diff
@@ -80,22 +90,57 @@ public void run() {
                      } catch (Exception e) {
                         log.keyMigrationFailed(Util.toStr(key), e);
                      }
-                  }
-               });
+                  });
+
+               }
+               es.shutdown();
+               try {
+                  while (!es.awaitTermination(500, TimeUnit.MILLISECONDS)) ;
+               } catch (InterruptedException e) {
+                  throw new CacheException(e);
+               }
+               return count.longValue();
+            }
+         } else {
+            final AtomicInteger count = new AtomicInteger(0);
+            ExecutorService executorService = Executors.newFixedThreadPool(threads);
```

Comment:
for consistency, use the same naming for the ExecutorService variable as above (es/executorService)